argparse_dataclass.py

```diff
@@ -320,102 +321,113 @@ def parse_known_args(
     and list of remaining arguments.
     """
     parser = argparse.ArgumentParser()
-    _add_dataclass_options(options_class, parser)
+    add_dataclass_options(options_class, parser)
     namespace, others = parser.parse_known_args(args=args)
     kwargs = _get_kwargs(namespace)
     return options_class(**kwargs), others
 
 
-def _add_dataclass_options(
-    options_class: Type[OptionsType], parser: argparse.ArgumentParser
-) -> None:
-    if not is_dataclass(options_class):
-        raise TypeError("cls must be a dataclass")
-
-    for field in fields(options_class):
-        args = field.metadata.get("args", [f"--{field.name.replace('_', '-')}"])
-        positional = not args[0].startswith("-")
-        kwargs = {
-            "type": field.metadata.get("type", field.type),
-            "help": field.metadata.get("help", None),
-        }
-
-        if field.metadata.get("args") and not positional:
-            # We want to ensure that we store the argument based on the
-            # name of the field and not whatever flag name was provided
-            kwargs["dest"] = field.name
+def field_to_argument_args(field: Field[Any]) -> Tuple[List[str], Dict[str, Any]]:
```

Comment:
Does this function need to be public? If so then I think a less confusing name would be better.

Reply:
Yes, public would be awesome. Maybe get_argument_args_from_dataclass_field? Open to any other suggestion.

Reply:
Done!